tests/php/Control/DirectorTest.php

```diff
@@ -100,30 +100,30 @@ public function testAbsoluteURL()
             //test empty / local urls
             foreach (['', './', '.'] as $url) {
                 $this->assertEquals("http://www.mysite.com:9090/mysite{$slash}", Director::absoluteURL($url, Director::BASE));
-                $this->assertEquals("http://www.mysite.com:9090{$slash}", Director::absoluteURL($url, Director::ROOT));
+                $this->assertEquals("http://www.mysite.com:9090/", Director::absoluteURL($url, Director::ROOT));
```

Comment:
Why did you change this? And why _didn't_ you change the line above?

Reply:
It's because of this line

`Director::config()->set('alternate_base_url', 'http://www.mysite.com:9090/mysite/');`

Note that the alternate_base_url is a subdir of mysite.com

This means that the the top-level mysite.com is treated like a completely different website e.g. an external link, so it's excluded from getting the add_trailing_slash logic

Reply:
All of these URLs use the alternate base URL, though?

> `$this->assertEquals("http://www.mysite.com:9090/mysite{$slash}"`
> 
> `$this->assertEquals("http://www.mysite.com:9090/"`

These have the exact same base URL as one another, both of which match the `alternate_base_url` and shouldn't be seen as external.

Reply:
`$this->assertEquals("http://www.mysite.com:9090/mysite{$slash}"`

The `{$slash}` means it IS being tested that it's adding or removing the slash i.e. it is being treated as the internal (not external) site 

`$this->assertEquals("http://www.mysite.com:9090/"` lacks the `/mysite` suffix - so it's an external site

Reply:
I've just tested the following:

```php
<?php

use SilverStripe\Control\Director;

Director::config()->set('alternate_base_url', 'http://www.mysite.com:9090/mysite/');
var_dump(Director::is_site_url('http://www.mysite.com:9090/')); // prints bool(true)
```

This shows that `http://www.mysite.com:9090/` is treated as internal by Director.

Reply:
Stepping through with a debugger, it's being caught by `if ($extension && $url !== Director::config()->get('alternate_base_url')) {`